### PR TITLE
add pull request guidelines

### PR DESCRIPTION
**Pull request by 0xc0170**
Starting as a base, I added some of the guidelines.

The main areas:
- clear git history
- useful commit messages
- address comments during review and get back to reviewers

Anything else should we add ?

### CODE REVIEW

**Comment by evedon** [comment]
I agree with the content but it would save you time to copy & paste from https://os.mbed.com/docs/mbed-os/v6.11/contributing/workflow.html where applicable

**Comment by 0xc0170** [comment]
> I agree with the content but it would save you time to copy & paste from https://os.mbed.com/docs/mbed-os/v6.11/contributing/workflow.html where applicable

This started as copy-paste, but I found some overlaps and some were not relevant here.
I simplified Mbed OS one for this project and rephrased some items. 

Let's compare to the Mbed OS PR guidelines and add/change this if it's better there.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -31,3 +31,11 @@ See the License for the specific language governing permissions and limitations
 Each commit must have at least one `Signed-off-by:` line from the committer to certify that the contribution is made under the terms of the [Developer Certificate of Origin](./DCO.txt).
 
 If you set your `user.name` and `user.email` as part of your git configuration, you can sign your commit automatically with `git commit -s`. You have to use your real name (i.e., pseudonyms or anonymous contributions cannot be made). This is because the DCO is a legally binding document.
+
+## Pull request guidelines
+
+- concise git commit messages (we recommend following Chris Beams 7 rules https://chris.beams.io/posts/git-commit/#seven-rules)
```
Comment (evedon) [inline CONTRIBUTING.md:37]: I propose the following:

```suggestion
- Git commit messages should be concise and must contain a proper description of the commit content. We recommend following Chris Beams 7 rules https://chris.beams.io/posts/git-commit/#seven-rules.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -31,3 +31,11 @@ See the License for the specific language governing permissions and limitations
...
+- add a prefix to your commit message to highlight the area of relevance. Write prefixes in lowercase, followed by a colon, followed by a space, and then finally the commit title
```
Comment (evedon) [inline CONTRIBUTING.md:38]:
```suggestion
- Add a prefix to your commit message to highlight the area of relevance. Write prefixes in lowercase, followed by a colon, followed by a space, and then finally the commit title.
```

Comment (evedon) [inline CONTRIBUTING.md:38]: Does it need to be lowercase? I see that this is not followed, most prefixes are either all uppercase or first letter uppercase.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -31,3 +31,11 @@ See the License for the specific language governing permissions and limitations
...
+- linear git history - no merge commits allowed, always rebase your branch
```
Comment (evedon) [inline CONTRIBUTING.md:39]:
```suggestion
- Keep a linear git history: no merge commits are allowed, always rebase your branch.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -31,3 +31,11 @@ See the License for the specific language governing permissions and limitations
...
+- preserve the history until the review is completed. Use `git commit --fixup` to add fixes. Once review is completed, run `git rebase --autosquash` to clean the history
```
Comment (evedon) [inline CONTRIBUTING.md:40]:
```suggestion
- Preserve the history until the review is completed. Use `git commit --fixup` to add fixes. Once review is completed, run `git rebase --autosquash` to clean the history.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -31,3 +31,11 @@ See the License for the specific language governing permissions and limitations
...
+- every new addition to already opened pull request should be followed by a comment to update reviewers what has changed
```
Comment (evedon) [inline CONTRIBUTING.md:41]:
```suggestion
- Every new addition to an already opened pull request should be commented to inform the reviewers on what has changed.
```

**Review by evedon — COMMENTED** [review]
Some edit suggestions and one question on prefix format.

File: CONTRIBUTING.md (hunk 2 above)
Comment (0xc0170) [reply CONTRIBUTING.md:38]: I'll change it to the first letter uppercase based on the history here.

**Comment by 0xc0170** [comment]
Commit suggestions via Github does not add signoff 🙄 

All comments were addressed.

**Comment by 0xc0170** [comment]
Conflict resolved via a rebase

**Review by MarceloSalazar — APPROVED** [review]
This looks good to me as initial guidelines.

**Review by evedon — APPROVED** [review]

**Pull request merged**